packages/core/src/modules/credentials/services/CredentialService.ts

```diff
@@ -247,16 +247,25 @@ export abstract class CredentialService {
     return this.credentialRepository.findById(connectionId)
   }
 
-  public async deleteById(credentialId: string, options?: DeleteCredentialOptions): Promise<void> {
-    const credentialRecord = await this.getById(credentialId)
+  /**
+   * Delete a credential exchange record from the repository and (optionally) 
+   * the wallet.
+   * Note that the default is now to delete from the wallet
+   * 
+   * @param credentialRecordId The id of the credential record
+   * @param options contains flag for determining whether to delete from wallet as well as repository
+
+   */
+  public async deleteById(credentialRecordId: string, options?: DeleteCredentialOptions): Promise<void> {
+    const credentialRecord = await this.getById(credentialRecordId)
 
     await this.credentialRepository.delete(credentialRecord)
 
-    if (options?.deleteAssociatedCredentials) {
-      for (const credential of credentialRecord.credentials) {
-        const formatService: CredentialFormatService = this.getFormatService(credential.credentialRecordType)
-        await formatService.deleteCredentialById(credentialRecord, options)
-      }
+    const deleteOptions = options ?? { deleteAssociatedCredentials: true }
```

Comment:
After thinking about this a bit more, this will break when more config options are added. We should probably do something like this:

``` 
  public async deleteById(credentialRecordId: string, { deleteAssociatedCredentials = true }: DeleteCredentialOptions = {}): Promise<void> {
    const credentialRecord = await this.getById(credentialRecordId)
```

Because currently if you call it like this (with an empty object):

```ts
agent.credentials.deleteById('id', {})
```

the value won't be set to true because options exist and the associated credential won't be deleted